vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -117,29 +101,31 @@ public synchronized ConsumableResourceResponse request(
               + "values should be one of the following: %s", workflowName, workflowPriority,
               acceptedPriorities.stream().map(String::valueOf).collect(Collectors.joining(", "))));
     }
+    if (workflows.get(workflowName) == null) {
+      set(workflowName, vidarrId, input);
+    }
+
     final var state = workflows.get(workflowName);
-    if (state == null) {
-      return ConsumableResourceResponse.error(
-          String.format("Internal Vidarr error: The %s workflow run priority has not been configured properly.", workflowName));
+
+    if (state == null || state.waiting.isEmpty()) {
```

Comment:
`set` adds the workflow to `workflows` if it doesn't yet exist, then we `get` the workflow from `workflows`, so in what case would `state` be null by this line?

Reply:
Taking a look, there really isn't a case, it's a redundant check after set() is called

Reply:
I think you could get rid of the first `set` and only `set` if the workflow run shouldn't launch.